code/modules/hydroponics/plant_genes.dm

```diff
@@ -200,71 +200,83 @@
 
 /// Traits that affect the grown product.
 /datum/plant_gene/trait
+	/// The rate at which this trait affects something.
 	var/rate = 0.05
+	/// Bonus lines displayed on examine.
 	var/examine_line = ""
-	/// Must be set and equal for any two traits of the same type
+	/// Traits that share this ID cannot be placed on the same plant.
 	var/trait_id
 	/// Flags that modify the final product.
 	var/trait_flags
+	/// A blacklist of seeds that a trait cannot be attached to.
+	var/list/obj/item/seeds/seed_blacklist
 
 /datum/plant_gene/trait/Copy()
 	var/datum/plant_gene/trait/G = ..()
 	G.rate = rate
 	return G
 
-/datum/plant_gene/trait/can_add(obj/item/seeds/S)
+/*
+ * Checks if we can add the trait to the seed in question.
+ *
+ * source_seed - the seed genes we're adding the trait too
+ */
+/datum/plant_gene/trait/can_add(obj/item/seeds/source_seed)
 	if(!..())
 		return FALSE
 
-	for(var/datum/plant_gene/trait/R in S.genes)
-		if(trait_id && R.trait_id == trait_id)
+	if(LAZYLEN(seed_blacklist))
+		for(var/obj/item/seeds/found_seed as anything in seed_blacklist)
+			if(istype(source_seed, found_seed))
+				return FALSE
```

Comment:
```suggestion
	for(var/obj/item/seeds/found_seed as anything in seed_blacklist)
		if(istype(source_seed, found_seed))
			return FALSE
```
No need to check for list length. `for (thing in null)` will simply not execute, as it's expected.